src/engine/controls/bpmcontrol.h

```diff
@@ -163,8 +169,8 @@ class BpmControl : public EngineControl {
     double m_dLastSyncAdjustment;
     bool m_dUserTweakingSync;
 
-    // m_pBeats is written from an engine worker thread
-    mixxx::BeatsPointer m_pBeats;
```

Comment:
Please revert. We need a direct reference to the constant BeatsPointer. With guramteed matching data. The indirection via track object does not work.

This also reveals, that we need a track independent sample rate in the beats object. 
Only this must be use, because only this gurantees to give the right sample value from the time based values. 

An alternative would be to convert all values to samples when importing the data from the protobuf.

Reply:
The persistent, time-based beats object does not need a sample rate. Only the frame-based wrapper at runtime does. This fact that has never been up for debate so nothing new is "revealed" here.